### ISSUE TITLE
Reading /dev/log

### ISSUE THREAD

**Issue by splitice** [vectordotdev/vector#10551]
Source:
```
[sources.syslog_unix]
type = "syslog"
max_length = 102_400
mode = "unix"
path = "/dev/log"
```

Result:
```
Dec 22 06:46:07.263  INFO transform{component_kind="transform" component_id=parse_logs component_type=remap component_name=parse_logs}: vrl_stdlib::log: Internal log [{ "facility": "authpriv", "host": "sudo", "hostname": "sudo", "message": "PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory
<83>Dec 22 06:46:03 sudo: PAM adding faulty module: pam_ldap.so
<83>Dec 22 06:46:04 sudo: pam_unix(sudo:auth): conversation failed
<82>Dec 22 06:46:04 sudo: pam_unix(sudo:auth): auth could not identify password for [**]
<81>Dec 22 06:46:04 sudo:   zabbix : user NOT in sudoers ; TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/bash **
", "severity": "err", "source_type": "syslog", "timestamp": t'2021-12-22T06:46:03Z' }] is being rate limited.
```

Not seperating by newline?

**Comment by spencergilbert** [comment]
I wonder if we're not getting a [`0xA`](https://vector.dev/docs/reference/configuration/sources/syslog/#line-delimiters), you could check that manually - switch to the `socket` source where there's more options available around [`framing`](https://vector.dev/docs/reference/configuration/sources/socket/#framing)

**Comment by jszwedko** [comment]
@splitice could you:

* share the full config here
* Vector version
* Expected behavior
* Actual behavior

From the description above it is unclear what you are expecting vs. what is happening.

**Comment by jszwedko** [comment]
The log message I see is from a `remap` transform that isn't present in the above config.

**Comment by splitice** [comment]
Of course remap: `log(.)` was used to troubleshoot the output. I suppose I could have used the prink sink.

We are talking about `/dev/log` here that is the output of the linux kernels `syslog()`. Users can't control its format.

I can see that there is some delimination in the output in that the message spans multiple lines, but none the less it should be easy enough to see in a select output from strace (I havent fixed the faulty PAM setup yet, so easy to replicate with sudo)

```
# strace -p 20089 -ff -s 100000 2>&1 | grep -i ldap | grep recv
[pid 20098] <... recvfrom resumed> "<83>Dec 23 00:43:32 sudo: PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory\0<83>Dec 23 00:43:32 sudo: PAM adding faulty module: pam_ldap.so\0", 8192, 0, NULL, NULL) = 210
[pid 20091] <... recvfrom resumed> "<83>Dec 23 00:43:32 su: PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory\0<83>Dec 23 00:43:32 su: PAM adding faulty module: pam_ldap.so\0<37>Dec 23 00:43:32 su: (to root) splitice on pts/1\0", 8192, 0, NULL, NULL) = 258
[pid 20097] <... recvfrom resumed> "<83>Dec 23 00:43:34 sudo: PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory\0<83>Dec 23 00:43:34 sudo: PAM adding faulty module: pam_ldap.so\0", 8192, 0, NULL, NULL) = 210
[pid 20094] recvfrom(31, "<83>Dec 23 00:43:35 sudo: PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory\0<83>Dec 23 00:43:35 sudo: PAM adding faulty module: pam_ldap.so\0", 8192, 0, NULL, NULL) = 210
[pid 20091] recvfrom(31, "<83>Dec 23 00:43:36 sudo: PAM unable to dlopen(pam_ldap.so): /lib/security/pam_ldap.so: cannot open shared object file: No such file or directory\0<83>Dec 23 00:43:36 sudo: PAM adding faulty module: pam_ldap.so\0", 8192, 0, NULL, NULL) = 210
```

It looks like /dev/log uses a single recv syscall response per message, and deliminates them with \0.

Would it be more correct to use the `socket` source, and vrl `parse_syslog`?

**Comment by jszwedko** [comment]
Ah, I see. If it is `\0` delimited, then the `syslog` source won't work since it expects `\n` delimited or octet-framed. That could be a nice enhancement though.

In your case, I think you could use the `socket` source with:

```toml
mode = "unix_stream"
framing.method = "character_delimited"
framing.character_delimited.delimiter = "\0"
```

And then use `parse_syslog` in a subsequent `remap` transform to parse the message contents as a syslog message.

**Comment by filterfish** [comment]
> In your case, I think you could use the socket source with:
> ```toml
> mode = "unix_stream"
> framing.method = "character_delimited"
> framing.character_delimited.delimiter = "\\0"
> ```

I'm trying to the same things as @splitice so I put the above in my config but I keep getting the following error message:

```
2022-01-12T06:38:33.833319Z ERROR vector::cli: Configuration error. error=invalid type: string "\\0", expected u8 for key `sources.syslog-local` at line 64 column 1
```

Any idea what I'm doing wrong?

I've also tried `\\t`, `\\n` `'\\u0000` and pretty much every other combination I can think of but they all result in the same error message.

My full config is:

```toml
[sources.syslog-local]
type = "socket"

host_key = "host"
mode = "unix_stream"
path = "/dev/log"

framing.method = "character_delimited"
framing.character_delimited.delimiter = '\0'

```
If I remove the two `framing.` entries then I get log messages as expected just not framed properly.

and I'm using version `vector 0.19.0 (x86_64-unknown-linux-gnu da60b55 2021-12-28)`

**Comment by jszwedko** [comment]
I think this is fixed by https://github.com/vectordotdev/vector/pull/10829 which will be released as 0.19.1 shortly. After that, you should be able to specify `framing.character_delimited.delimiter = "\u0000"`
